internal/pkg/docker/dockerfile/dockerfile.go

```diff
@@ -93,27 +111,56 @@ func (df *Dockerfile) parse() error {
 	}
 	defer file.Close()
 
-	scanner := bufio.NewScanner(file)
-	parsedDockerfile := parseFromScanner(scanner)
+	f, err := ioutil.ReadFile(file.Name())
+	if err != nil {
+		return fmt.Errorf("dockerfile readfile error: %w %s", err, f)
+	}
+	fileString := string(f)
+
+	parsedDockerfile, err := parseFromReader(fileString)
+	if err != nil {
+		return fmt.Errorf("dockerfile parse error: %w", err)
+	}
 
 	df.ExposedPorts = parsedDockerfile.ExposedPorts
+	df.HealthCheck = parsedDockerfile.HealthCheck
 	df.parsed = true
 	return nil
 }
 
-func parseFromScanner(scanner *bufio.Scanner) Dockerfile {
-	var line = ""
+func parseFromReader(line string) (Dockerfile, error) {
 	var df Dockerfile
 	df.ExposedPorts = []portConfig{}
-	for scanner.Scan() {
-		line = scanner.Text()
-		switch {
-		case strings.HasPrefix(line, "EXPOSE"):
-			currentPorts := parseExpose(line)
+	df.HealthCheck = healthCheck{}
+
+	// Parse(rwc io.Reader) reads lines from a Reader, and parses the lines into an AST.
+	ast, err := parser.Parse(strings.NewReader(line))
+	if err != nil {
+		return df, err
+	}
+	for i := 0; i < len(ast.AST.Children); i++ {
+		// ParseInstruction converts an AST to a typed instruction.
+		// Example of an instruction is HEALTHCHECK CMD curl -f http://localhost/ || exit 1.
+		instruction, err := instructions.ParseInstruction(ast.AST.Children[i])
+		if err != nil {
+			return df, err
+		}
+		inst := fmt.Sprint(instruction)
```

Comment:
What do you think of `instruction.String()` instead? Since under the hood it is calling `String()` method.

Reply:
This gives me an undefined error